File: docs/public/specs/esm/density-groups.js
import * as vg from "@uwdata/vgplot";

await vg.coordinator().exec([
  vg.loadParquet("penguins", "data/penguins.parquet")
]);

const $bandwidth = vg.Param.value(20);
const $normalize = vg.Param.value("none");
const $stack = vg.Param.value(false);
const $offset = vg.Param.value(null);

export default vg.vconcat(
  vg.hconcat(
    vg.menu({label: "Normalize", as: $normalize, options: ["none", "sum", "max"]}),
    vg.menu({label: "Stack", as: $stack, options: [false, true]}),
    vg.menu({
      label: "Offset",
      as: $offset,
      options: [
      {label: "none", value: null},
      {label: "normalize", value: "normalize"},
      {label: "center", value: "center"}
    ]
    })
  ),
  vg.plot(
    vg.densityY(
      vg.from("penguins"),
      {
        x: "bill_depth",
        fill: "species",
        fillOpacity: 0.4,
        bandwidth: $bandwidth,
        normalize: $normalize,
        stack: $stack,
        offset: $offset
      }
    ),
    vg.marginLeft(50),
    vg.height(200)
  )
);

File: docs/public/specs/python/density-groups.py
import vgplot as vg

penguins = vg.parquet("data/penguins.parquet")

bandwidth = vg.param(20)
normalize = vg.param("none")
stack = vg.param(False)
offset = vg.param(None)

view = vg.vconcat(
    vg.hconcat(
        vg.menu(label="Normalize", bind=normalize, options=["none", "sum", "max"]),
        vg.menu(label="Stack", bind=stack, options=[False, True]),
        vg.menu(
            label="Offset",
            bind=offset,
            options=[vg.option("none", value=None), "normalize", "center"],
        ),
    ),
    vg.plot(
        vg.density_y(
            penguins,
            x="bill_depth",
            fill="species",
            fill_opacity=0.4,
            bandwidth=bandwidth,
            normalize=normalize,
            stack=stack,
            offset=offset,
        ),
        vg.margin_left(50),
        vg.height(200),
    ),
)
